src/cmds/scripts/post_install.py

```diff
@@ -0,0 +1,463 @@
+		replace(pg_hba_file_path,original_str,replacement_str)
+		print "pg_hba file Successfully configured"
+	else:
+		print "pg_hba file is not available to configure"
+		exit(6)
+		
+def InstallDataService(username, password):
+	#print pbs_path
+	pbs_pathquote="\""+pbs_path
+	#print pbs_pathquote
+	newpath=pbs_path + 'home\\server_priv\\'
+	if not os.path.exists(newpath):
+		os.makedirs(newpath)
+	# Creation of db_user file at PBS_HOME\server_priv directory
+	dbuserfile=newpath+ 'db_user'
+	ErrChk="IF %ERRORLEVEL% NEQ 0 GOTO ERR\r\n"
+	file = open(dbuserfile,"w") 
+	file.write(username) 
+	file.close() 
+	# Creation of createdb.bat script at PBS_EXEC\etc directory
+	DBpath = pbs_pathquote + "home\\datastore\""
+	createdbfile=pbs_path + 'exec\\etc\\createdb.bat'
+	#print createdbfile
+	file = open(createdbfile,"w")  
+	buffer="@echo off \r\n"
+	file.write(buffer) 
+	# echo PBS_CONF_FILE
+	buffer="set PBS_CONF_FILE=" + pbs_path + "pbs.conf\r\n"
+	file.write(buffer)
+	# initialize database cluster
+	buffer=pbs_pathquote + "exec\\pgsql\\bin\\initdb.exe\"" + " -D " + DBpath + " -E SQL_ASCII --locale=C\r\n"
+	file.write(buffer)
+	file.write(ErrChk)
+	# start database
+	buffer="call " + pbs_pathquote + "exec\\sbin\\pbs_dataservice\"" + " start\r\n"
+	file.write(buffer)
+	file.write(ErrChk)
+	# create db
+	buffer=pbs_pathquote+"exec\\pgsql\\bin\\createdb.exe\" -p 15007 pbs_datastore\r\n"
+	file.write(buffer)
+	file.write(ErrChk)
+	# install schema
+	buffer=pbs_pathquote+"exec\\pgsql\\bin\\psql.exe\" -p 15007 -d pbs_datastore -f "+pbs_pathquote+"exec\\etc\\pbs_db_schema.sql\"\r\n"
+	file.write(buffer)
+	file.write(ErrChk)
+	# create superuser 
+	buffer=pbs_pathquote+"exec\\pgsql\\bin\\psql.exe\" -p 15007 -d pbs_datastore -c \"create user \\\"" + username + "\\\" SUPERUSER LOGIN\"\r\n"
+	file.write(buffer)
+	# run Pbs_ds_password
+	buffer=pbs_pathquote+"exec\\sbin\\pbs_ds_password.exe\" -r\r\n"
+	file.write(buffer)
+	file.write(ErrChk)
+	# stop database
+	buffer="call "+ pbs_pathquote + "exec\\sbin\\pbs_dataservice\"" + " stop\r\n"
+	file.write(buffer)
+	file.write(ErrChk)
+	# error handling
+	file.write("exit /b 0\r\n")
+	file.write(":ERR\r\n")
+	# stop database
+	buffer="call "+ pbs_pathquote + "exec\\sbin\\pbs_dataservice\"" + " stop\r\n"
+	file.write(buffer)
+	file.write("exit /b 1\r\n")
+	file.close()
+	print "Executing createdbfile as install user"
+	with open(os.devnull, 'w') as fd:
+			theproc= subprocess.Popen([createdbfile], stdout=fd, stderr=fd)
+			theproc.communicate()
+			#print theproc.returncode
+			if theproc.returncode > 0:
+				print "Failed to initilize database"
+				sys.exit(3)
+			else:
+				print "Database intializaation success"
+				
+	UpdateDSconfig()
+	print "InstallDataService completed... "
+	
+	
+def CreateServerdbFile():
+	#os.system("net stop pbs_server")
+	with open(os.devnull, 'w') as fd:
+			theproc= subprocess.Popen(["sc", "stop", "pbs_server"], stdout=fd, stderr=fd)
```

Comment:
I think you need to use 'net stop pbs_server' instead 'sc stop pbs_server' because sc is unblocking command, it just send stop request to server and returns while net wait for service to stop before returning